From 0173ce2a54a3fbd0a3d4db42dc7e2f7490a30404 Mon Sep 17 00:00:00 2001
Message-Id: <0173ce2a54a3fbd0a3d4db42dc7e2f7490a30404.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Scott D Phillips <scott.d.phillips@intel.com>
Date: Tue, 16 Sep 2014 00:05:04 +0000
Subject: [PATCH 256/312] REVERTME [VPG]: Update the primary plane on zorder
 change

When the zorder is updated the primary plane also needs to be
updated as well in case it needs to update its alpha.

The whole primary plane alpha optimization goop is slated for
removal when gralloc/hwc superceed it with a userspace version.

Issue: GMINL-1364
Change-Id: I125fc6b916b3ccd0f908872f324c938158a1ed11
Signed-off-by: Scott D Phillips <scott.d.phillips@intel.com>
---
 drivers/gpu/drm/i915/intel_sprite.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_sprite.c b/drivers/gpu/drm/i915/intel_sprite.c
index d631527..3bd8dc2 100644
--- a/drivers/gpu/drm/i915/intel_sprite.c
+++ b/drivers/gpu/drm/i915/intel_sprite.c
@@ -319,6 +319,7 @@ int i915_set_plane_zorder(struct drm_device *dev, void *data,
 
 	struct intel_crtc *intel_crtc =
 			to_intel_crtc(dev_priv->plane_to_crtc_mapping[pipe]);
+	struct drm_crtc *crtc = &intel_crtc->base;
 
 	s1_zorder = (order >> 3) & 0x1;
 	s1_bottom = (order >> 2) & 0x1;
@@ -373,6 +374,10 @@ int i915_set_plane_zorder(struct drm_device *dev, void *data,
 		intel_crtc->sprite1_alpha = true;
 	else
 		intel_crtc->sprite1_alpha = false;
+
+	dev_priv->display.update_primary_plane(crtc, crtc->primary->fb, crtc->x,
+		crtc->y);
+
 	return 0;
 }
 
-- 
1.7.9.5

